packages/material-ui/src/SwipeableDrawer/SwipeableDrawer.js

```diff
@@ -201,6 +278,19 @@ const SwipeableDrawer = React.forwardRef(function SwipeableDrawer(props, ref) {
     const currentX = calculateCurrentX(anchorRtl, event.touches);
     const currentY = calculateCurrentY(anchorRtl, event.touches);
 
+    if (open && paperRef.current.contains(event.target)) {
```

Comment:
What do you think of introducing a `nodeWhoClaimedTheScroll` like variable back? I think that we can improve the UX by disabling the swipe for all the touch session if we have found one native scroll handler. Does it make sense, I can provide more detail if needed.

Reply:
By not doing the touch start, everything works just fine. I don't see why introducing a global lock would help.

Reply:
I have observed two issues using this logic:

1. Once you reach the end of the native scroll, the drawer starts to drag. If you look at the behavior of nested scroll containers, you would need to reach the end of the scrollbar with the nested container and start a new interaction to scroll the parent. In this demo, I don't think that the drawer should start to drag. It's the behavior implemented react-swipeable-views, I think that it's more intuitive/less disturbing.

![double interactions](https://user-images.githubusercontent.com/3165635/67469150-131b1480-f64c-11e9-872c-376c1fded21f.gif)

2. Because of 1., something else happens:
![jump](https://user-images.githubusercontent.com/3165635/67469538-a18f9600-f64c-11e9-9b7d-041bc6cbab65.gif)

Reply:
> In this demo, I don't think that the drawer should start to drag.

I liked that behavior, but I see why it may not be desired. Regarding point 2: oh... Allright, 1. should fixed.

Reply:
Regarding 1. I have a strong conviction that we should allow a single scroll container, per touch session (start, move, end).

I'm no longer sure that 2. is strictly related to 1. I think that it requires further investigation.

Reply:
I fixed both issues by re-introducing `nodeThatClaimedTheSwipe`. It isn't used for inter-drawer locking anymore, but for drawer vs. scrollable container inside that drawer locking, that's why it's updating while moving (to fix the behavior in your gif).